emcc.py

```diff
@@ -2556,14 +2556,14 @@ def do_binaryen(target, asm_target, options, memfile, wasm_binary_target,
       shutil.copyfile(wasm_binary_target, os.path.join(shared.get_emscripten_temp_dir(), 'pre-eval-ctors.wasm'))
     shared.Building.eval_ctors(final, wasm_binary_target, binaryen_bin, debug_info=debug_info)
   # after generating the wasm, do some final operations
-  if not shared.Settings.WASM_BACKEND:
-    if shared.Settings.SIDE_MODULE:
+  if shared.Settings.SIDE_MODULE:
+    if not shared.Settings.WASM_BACKEND:
       wso = shared.WebAssembly.make_shared_library(final, wasm_binary_target)
       # replace the wasm binary output with the dynamic library. TODO: use a specific suffix for such files?
       shutil.move(wso, wasm_binary_target)
       if not DEBUG:
         os.unlink(asm_target) # we don't need the asm.js, it can just confuse
-      sys.exit(0) # and we are done.
+    sys.exit(0) # and we are done.
```

Comment:
So we now exit 0 if SIDE_MODULE and WASM_BACKEND, where before we would ignore SIDE_MODULE and do compilation normally?
Neither of those seem great, we're silently not doing what the user asked.
How about change the `if not WASM_BACKEND` to `assert not WASM_BACKEND`? Or is there some other reason we don't do that

Reply:
Or if we disallow SIDE_MODULE and WASM_BACKEND from being set together in Settings itself (at the end of this diff, in shared.py), do we need the `if not WASM_BACKEND` here at all?

Reply:
This was my first attempt at gettting SIDE_MODULE to work with WASM_BACKEND.  Basically we want to early exit here when SIDE_MODULE is used so we don't go onto great .js stuff.   The make_shared_library() call doesn't currently work with wasm backend, but we exiting is still better than continuing.  I could just revert this.. but actually I think its best to remove the WASM_BACKEND check here instead, and left verify_settings() to the checking..